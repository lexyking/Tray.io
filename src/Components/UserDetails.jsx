import React, { Component, Fragment } from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

class userDetails extends Component {

  //Function continue to move to the next tab
  continue = e => {
    e.preventDefault();
    this.props.nextStep();
  }


  render() {
    //Destructuring attributes from the props
    const { values, handleChange } = this.props
    return (
      <MuiThemeProvider>
        <section className="mainSection">
          <section className="formSection">
            <h2>Enter User Details</h2>
            <TextField
              hintText="Enter Your Name"
              floatingLabelText="Name *"
              onChange={handleChange('name')}
              defaultValue={values.name}
            />
            <br />
            <TextField
              hintText="Enter Your Role"
              floatingLabelText="role *"
              onChange={handleChange('role')}
              defaultValue={values.role}
            />
            <br />
            <TextField
              hintText="Enter Your Email"
              floatingLabelText="Email *"
              onChange={handleChange('email')}
              defaultValue={values.email}
            />
            <br />
            <TextField
              hintText="Enter Your Password"
              floatingLabelText="Password *"
              onChange={handleChange('password')}
              defaultValue={values.password}
            />
            <br />
            <RaisedButton
              label="continue"
              primary={true}
              style={styles.button}
              onClick={this.continue}
            />
          </section>
        </section>
      </MuiThemeProvider>
    )

  }
}

const styles = {
  button: {
    margin: 15
  }
}

export default userDetails
