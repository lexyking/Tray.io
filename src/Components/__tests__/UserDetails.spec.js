import React from 'react';
import { create } from 'react-test-renderer';
import UserDetails from '../UserDetails';


describe('UserDetails component', () => {
  test('Render Correctly', () => {
    const userDetails = create(<UserDetails values />);
    expect(userDetails.toJSON()).toMatchSnapshot();
  })
})