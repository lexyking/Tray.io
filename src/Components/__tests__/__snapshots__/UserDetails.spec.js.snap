// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails component Render Correctly 1`] = `
Array [
  "",
  <section
    className="mainSection"
  >
    <section
      className="formSection"
    >
      <h2>
        Enter User Details
      </h2>
      <article
        className="inputFields"
      >
        <label>
          Name
          <span
            className="asterisk"
          >
            *
          </span>
        </label>
        <input
          name="name"
          placeholder="Enter Your Name"
          required={true}
          type="text"
        />
      </article>
      <article
        className="inputFields"
      >
        <label>
          Role
          <span
            className="asterisk"
          >
            *
          </span>
        </label>
        <input
          name="role"
          placeholder="Enter Your Role"
          type="text"
        />
      </article>
      <article
        className="inputFields"
      >
        <label>
          Email
          <span
            className="asterisk"
          >
            *
          </span>
        </label>
        <input
          name="email"
          placeholder="Enter Your Email"
          type="text"
        />
      </article>
      <article
        className="inputFields"
      >
        <label>
          Password
          <span
            className="asterisk"
          >
            *
          </span>
        </label>
        <input
          name="password"
          placeholder="Enter Your Password"
          type="password"
        />
      </article>
      <button
        className="btn"
        onClick={[Function]}
      >
        Continue
      </button>
    </section>
  </section>,
]
`;
